here-naksha-lib-psql/src/commonMain/kotlin/naksha/psql/executors/write/DropCollection.kt

```diff
@@ -0,0 +1,51 @@
+package naksha.psql.executors.write
+
+import naksha.base.Platform.PlatformCompanion.logger
+import naksha.base.PlatformLogger
+import naksha.model.*
+import naksha.model.Naksha.NakshaCompanion.VIRT_COLLECTIONS
+import naksha.model.Naksha.NakshaCompanion.VIRT_COLLECTIONS_QUOTED
+import naksha.model.objects.NakshaCollection
+import naksha.psql.*
+import naksha.psql.executors.WriteExt
+
+class DropCollection(private val session: PgSession) {
+
+    fun execute(map: PgMap, write: WriteExt): TupleNumber {
+        if(write.collectionId != VIRT_COLLECTIONS){
+            throw NakshaException(
+                NakshaError.ILLEGAL_ARGUMENT,
+                "Expected $VIRT_COLLECTIONS collectionId when dropping collections"
+            )
+        }
+        val collectionId = write.featureId ?: throw NakshaException(
+            NakshaError.ILLEGAL_ARGUMENT,
+            "DROP without collectionId (expected in write's 'featureId')"
+        )
+        val pgCollection = PgCollection(map, collectionId)
```

Comment:
Actually, this should not even be possible, the only way to ask for a PgCollection instance should be using the `map[collectionId]` method, the reason is concurrency, all threads share the same instances to improve state caching. Actually, the constructor is `internal`, and this code is in another package, I wonder why this code is even possible, did I miss something? We need to fix both, nobody must be able to invoke the constructor of `PgCollection` except for the internal code (maybe we make it protected and the extending classes internal?). Can you change this into:
`pgCollection = map[collectionId]`

Reply:
changed